src/lib/util.js

```diff
@@ -41,8 +40,53 @@ function getRejectedPromiseIfTimedOut (timeoutMs) {
   );
 }
 
+/**
+ * Wraps an onSuccess and onError handler around a function
+ * @param {function} orignalFn
+ * @param {function} [onError]
+ * @param {function} [onSuccess]
+ * @return {*} - returns whatever the original function would have returned
+ */
+const wrapCompletedHandlers = (orignalFn, onError, onSuccess) => (...args) => {
+  let returnVal;
+  const callOnSuccess = (returnVal) => {
+    if (isFunction(onSuccess)) {
+      onSuccess(returnVal, args);
+    }
+  };
+  const callOnError = (error) => {
+    if (isFunction(onError)) {
+      onError(error);
+    }
+  };
+
+  try {
+    returnVal = orignalFn(...args);
```

Comment:
^ Can originalFn be an async function? If so, this returns a promise, and doesn't throw if the promise rejects ya?  I think we want the original listernfn to be allowed to be async right?